src/explore-education-statistics-common/src/components/form/rhf/__tests__/RHFForm.test.tsx

```diff
@@ -0,0 +1,491 @@
+              <button
+                type="button"
+                onClick={() => {
+                  reset();
+                }}
+              >
+                Reset form
+              </button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    fireEvent.click(screen.getByText('Reset form'));
+
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Something went wrong'),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test('removes submit error when form values are changed', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: 'Firstname',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm
+              id="test-form"
+              showSubmitError
+              onSubmit={() => {
+                throw new Error('Something went wrong');
+              }}
+            >
+              <label htmlFor="firstName">Firstname</label>
+              <input
+                type="text"
+                id="firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.queryByText('Something went wrong')).toBeInTheDocument();
+    });
+
+    fireEvent.change(screen.getByLabelText('Firstname'), {
+      target: {
+        value: 'Another firstname',
+      },
+    });
+    await waitFor(() => {
+      expect(
+        screen.queryByText('Something went wrong'),
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test('focuses error summary on submit', async () => {
+    render(
+      <FormProvider
+        initialValues={{
+          firstName: '',
+        }}
+        validationSchema={Yup.object({
+          firstName: Yup.string().required(),
+        })}
+      >
+        {({ register }) => {
+          return (
+            <RHFForm id="test-form" showSubmitError onSubmit={noop}>
+              <input
+                id="test-form-firstName"
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...register('firstName')}
+              />
+              <button type="submit">Submit</button>
+            </RHFForm>
+          );
+        }}
+      </FormProvider>,
+    );
+
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => {
+      expect(screen.getByRole('alert')).toHaveFocus();
```

Comment:
Using `getByRole` in a `waitFor`. We could check on the error summary text instead, then check focus after the `waitFor`:

```ts
await waitFor(() => {
  expect(screen.getByText('There is a problem')).toBeInTheDocument();
});

expect(screen.getByRole('alert')).toHaveFocus();
```